### PR TITLE
#95-categoryDropdown

### LINKED ISSUE #95: (SP:2) Implement stepper Category selection
**User Story #13** 
## Description 

Implement **Category** selection with image at Stepper `Subjects` step

[Figma link
](https://www.figma.com/file/1NqYJ5oaby8tlU7zLQBOja/MVPSpace2Study-(Students)?type=design&node-id=1955-53423&mode=design&t=LfY6O3UorMwD5SQo-4)
Image

### PR DESCRIPTION
**Pull request by nadraharr**
Now user can choose his main category on first login.

### CODE REVIEW

**Review by AntonRomanov999 — APPROVED** [review]

**Review by zim89 — APPROVED** [review]

**Review by natamrshn — APPROVED** [review]

File: src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 
 const SubjectsStep = ({ btnsBox }) => {
   return (
     <Box sx={styles.container}>
+      <Box sx={styles.imgContainer}>
+        <Box component='img' src={img} sx={styles.img} />
+      </Box>
       <Box sx={styles.rigthBox}>
-        Subjects step
+        <CategoryDropdown />
```
Comment (melyasanya) [inline src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx]: I don't think this is a correct component to use here. At least not like this. Fro example, placeholder in the task should be Main tutoring category, not Category. Plus, in this case users should be able to create new categories. This component was created for different use, so you should create another dropdown or try adapt this one, if it is possible

**Review by melyasanya — CHANGES_REQUESTED** [review]

File: src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx (hunk 1 above)
Comment (nadraharr) [reply src/containers/tutor-home-page/subjects-step/SubjectsStep.jsx]: CategoryDropdown is adapted now. Please, check last code updates.

**Review by lublubczyk — APPROVED** [review]

**Review by VoitV — APPROVED** [review]

File: src/containers/category-dropdown/CategoryDropdown.tsx [hunk 2]
```diff
@@ -20,25 +20,29 @@ import {
...
-import { styles } from '~/containers/category-dropdown/CategoryDropdown.styles'
 
 interface CategoryDropdownInterface {
   category: string | null
   onCategoryChange: (
     _: SyntheticEvent,
     value: CategoryNameInterface | null
   ) => void
+  authorizedCreateCategory: boolean | null
+  label: string | null
```
Comment (melyasanya) [inline src/containers/category-dropdown/CategoryDropdown.tsx]: you can just make it optional, no need in lable being null

**Review by melyasanya — CHANGES_REQUESTED** [review]
Are you sure, that data in dropdown will be from the categories that we need in this component? Now it is fetching resource categories, but we need it to fetch subjects

File: src/containers/category-dropdown/CategoryDropdown.tsx (hunk 2 above)
Comment (nadraharr) [reply src/containers/category-dropdown/CategoryDropdown.tsx]: https://github.com/Project-Based-10-24-vk/Client/pull/125/commits/413a34dce34c9a713a5cd367b8cb915bc1b40133

**Comment by nadraharr** [comment]
> Are you sure, that data in dropdown will be from the categories that we need in this component? Now it is fetching resource categories, but we need it to fetch subjects

I relied on the description of the task: https://github.com/Project-Based-10-24-vk/Client/issues/95
The SubjectStep component will also have a separate dropdown for subjects.

**Review by melyasanya — APPROVED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**